### ISSUE TITLE
How to backup containers that depend on the parent container 

### ISSUE THREAD

**Issue by indomitorum** [Minituff/nautical-backup#164]
Debian Bookwork - Docker - NFS

The backup works well for single containers without dependencies. My question is, how do you backup the dependencies as well?

Example:

Container names with docker ps --format "{{.Names}}"
authentik-worker-1
authentik-server-1
authentik-postgresql-1
authentik-redis-1

There is a single authentik folder that has got the docker-compose.yml. Do I have to back them all up individually with the  override_source_dir?

Example of docker-compose.yml:

version: '3'
services:
  nautical-backup:
    image: minituff/nautical-backup:1.4
    container_name: nautical-backup
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /home/xxxxx:/app/source
      - /home/xxxxx/nfs_docker:/app/destination
    environment: # Optional variables
      - TZ=Europe/London
      - CRON_SCHEDULE=00 14 * * *  
      - OVERRIDE_SOURCE_DIR=nginx-app-1:nginx,authentik-server-1:authentik,authentik-worker-1:authentik,authentik-postgresql-1:authentik,authentik-redis-1:authentik
      - SKIP_CONTAINERS=homeassistant,esphome

They get backed up but I am not sure if:

1. this is correct
2. how would you restore thiscorrectly as some containers do have dependencies like databases etc.

Thanks

**Comment by Minituff** [comment]
Hey this is a good question. I don't think this is possible right now at the moment. This question was also brought up here: https://github.com/Minituff/nautical-backup/issues/88.

I do plan on implementing this feature soon because it's very important.

For now, I would recommend you `skip` and containers that have multiple components.

I'll post and update on this issue once I complete it.

**Comment by mixpc** [comment]
@indomitorum Thank you for bringing this topic to the forefront as I have the same case with two Wordpress containers:
```
wordpress-db-sitename
wordpress-sitename
```
referenced in the same docker-compose.yml under the same folder wordpress.

@Minituff Thank you for your interest. And I must say I've tried different docker backup options, and nautical-backup, to me, excels! I can't understand why so few stars.

**Comment by Minituff** [comment]
Thanks @mixpc! This is definitely the #1 feature that I'm looking to add next. I'm just waiting on some free time so that so I can begin developing.

**Comment by Minituff** [comment]
Hey, I added this feature in [Version 2.0.0](https://github.com/Minituff/nautical-backup/releases/tag/v2.0.0).

I am going to close this issue, but if this feature is not working or you would like some more features, feel free to reopen it or create a new issue.
